src/agent/internals/src/test/java/com/google/devtools/cdbg/debuglets/java/GceMetadataQueryTest.java

```diff
@@ -0,0 +1,167 @@
+package com.google.devtools.cdbg.debuglets.java;
+
+import static com.google.common.truth.Truth.assertThat;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLStreamHandler;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(JUnit4.class)
+public class GceMetadataQueryTest {
+
+  @Rule public final MockitoRule mocks = MockitoJUnit.rule();
+
+  @Mock HttpURLConnection projectIdUrlConnection;
+  @Mock HttpURLConnection projectNumberUrlConnection;
+  @Mock HttpURLConnection accessTokenUrlConnection;
+
+  private GceMetadataQuery metadataQuery;
+
+  @Before
+  public void setUp() throws MalformedURLException {
+    URL baseUrl =
+        new URL(
+            "mock",
+            "metadata.service",
+            80,
+            "/",
+            new URLStreamHandler() {
+              @Override
+              protected HttpURLConnection openConnection(URL url) throws IOException {
+                String base = "mock://metadata.service:80/";
+
+                if (url.toString().equals(base + "project/project-id")) {
+                  return projectIdUrlConnection;
+                }
+
+                if (url.toString().equals(base + "project/numeric-project-id")) {
+                  return projectNumberUrlConnection;
+                }
+
+                if (url.toString().equals(base + "instance/service-accounts/default/token")) {
+                  return accessTokenUrlConnection;
+                }
+
+                return null;
+              }
+            });
+
+    metadataQuery = new GceMetadataQuery(baseUrl);
+  }
+
+  @Test
+  public void environmentQuerySuccess() throws IOException {
+    when(projectIdUrlConnection.getInputStream()).thenReturn(streamFromString("best_project_ever"));
+    when(projectNumberUrlConnection.getInputStream()).thenReturn(streamFromString("3456273849"));
+
+    assertThat(metadataQuery.getProjectId()).isEqualTo("best_project_ever");
+    assertThat(metadataQuery.getProjectNumber()).isEqualTo("3456273849");
+  }
+
+  @Test
+  public void environmentQueryCache() throws IOException {
+    when(projectIdUrlConnection.getInputStream()).thenReturn(streamFromString("best_project_ever"));
```

Comment:
What line length are you using for formatting?  Just asking because it's at the threshold where this is wrapping but most code hasn't been.

Reply:
100, looks like this line happens to be right at that threshold.